docs/docs/api/Dispatcher.md

```diff
@@ -952,6 +952,44 @@ const client = new Client("http://example.com").compose(
 );
 ```
 
+##### `dump`
+
+The `dump` interceptor enables you to dump the response body from a request upon a given limit.
+
+**Options**
+- `maxSize` - The maximum size (in bytes) of the response body to dump. If the size of the request's body exceeds this value then the connection will be closed. Default: `1048576`.
+- `abortOnDumped` - States whether or not abort the request after the response's body being dumped. Default: `true`.
+- `waitForTrailers` - Hints the dispatcher to wait for trailers if the response's body has been dumped. Default: `false`.
```

Comment:
I would skip both of these. Also not sure what abortOnDumped does is needed for?

Reply:
Actually, what would make sense is to have a `dumpOnAbort` abort option.

Reply:
For `abortOnDumped` I was thinking for use cases where you do not want to abort the request but dump its body, in that way things calls to e.g. `client.request` does not reject the promise but rather returns you an empty `''` as body.

`waitForTrailers` in the case they are interested in the trailers, not really common so happy to drop it 👍 

For `dumpOnAbort`, still not sure, do you mean intercept the abort call and do not abort the request but rather consume it?

Reply:
>  I was thinking for use cases where you do not want to abort the request but dump its body, in that way things calls to e.g

I'm confused. When would you not want that?

> do you mean intercept the abort call and do not abort the request but rather consume it?

So if you call abort in a sub handler, that will not kill the connection, but try to consume it instead, in order to re-use the connection.

Reply:
>I'm confused. When would you not want that?

I can honestly just think in situations where you are just interested in dump it but do not want the request to be rejected (which currently happens if the request is aborted and try to read the body).

>So if you call abort in a sub handler, that will not kill the connection, but try to consume the body instead, that way one can re-use the connection of an aborted request.

Got it, that was the part that I was missing; then its not that far from what I've had in mind. Let me draft it quickly

Reply:
>So if you call abort in a sub handler, that will not kill the connection, but try to consume the body instead, that way one can re-use the connection of an aborted request.

Thinking twice, this was the original situation of `abortOnDumped`, meaning that if enabled we'll abort the request and drop the connection; if disabled, we consume the body and keep the connection alive.

wdyt if we go with `dumpOnAbort` and we make naturally the handle do not abort the request so the connection can be reused?
Basically what you suggested and by nature we make the handler do not drop connections, otherwise it kind of feel pointless the interceptor as we are dropping the connection on every dump